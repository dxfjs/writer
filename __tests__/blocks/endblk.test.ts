import { EndBlk, Handle, TagsManager } from "@/index";

describe("EndBlk class", () => {
  it("should create an endblk instance", () => {
    const block = new EndBlk(new Handle());
    block.addAppDefined("ACAD_REACTORS");
    const mg = new TagsManager();
    block.tagify(mg);
    expect(mg.stringify()).toMatchSnapshot();
  });

  it("should create defined application", () => {
    const block = new EndBlk(new Handle());
    const reactors = block.addAppDefined("ACAD_REACTORS");
    expect(reactors.name).toBe("ACAD_REACTORS");
    const test = block.addAppDefined("ACAD_REACTORS");
    expect(test).toBe(reactors);
  });
});
