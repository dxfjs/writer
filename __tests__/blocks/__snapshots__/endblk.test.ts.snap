// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EndBlk class > should create an endblk instance 1`] = `
"0
ENDBLK
5
1
330
0
100
AcDbEntity
8
0
100
AcDbBlockEnd"
`;
